blynktimer.py

```diff
@@ -44,8 +44,10 @@ def __init__(self, no_timers_err=True):
         self.no_timers_err = no_timers_err
 
     def _get_func_name(self, obj):
```

Comment:
My suggestion to leave your logic but modify code a little bit 

`    def _get_func_name(self, obj):
        if getattr(obj, '__name__', None) is None:
            return self._get_func_name(obj.func)
        elif getattr(obj, 'func', None) is None:
            return 'timer'
        return obj.__name__`

Reply:
Unfortunately this wouldn't work. If `obj` is of type `function` and `MICROPY_PY_FUNCTION_ATTRS` is disabled, then its only internal attribute is `__class__`,
so this will always call `obj.func` which will fail with an `AttributeError`

In fact, even with function attrs it only adds `__name__`, so I am assuming your attempting to check for `obj.func` in the case that you pass a decorator object to `_get_func_name` but unless I am missing something I only see where a `function` type object could be passed:
```python
 Timer.timers['{}_{}'.format(len(list(Timer.timers.keys())), blynk._get_func_name(func))] = _timer
```
where `func` comes directly from the decorator `__init__` method. Is this correct or am I overlooking something?

Reply:
If it is true that only a `function` object would be passed to `_get_func_name`, then it could be reduced to:
```python
def _get_func_name(self, obj):
    return getattr(obj, '__name__', 'timer')
```

Reply:
If you preffer hasattr - ok lets use them.
Here is my suggestion. You see we need return obj.func only if there is no __name__
```
def _get_func_name(self, obj):
    if hasattr(obj, 'func') is False:
        return 'timer'
    elif hasattr(obj, '__name__') is False:
        return self._get_func_name(obj.func)
    return obj.__name__
```

Reply:
Okay, I still don't see what `obj.func` could ever be, cause as far I see, the only attributes `obj` could have are: `__class__` and possibly `__name__`

I uploaded blynktimer.py with the above snippet to an ESP32 which is function attributes enabled, so it has the `__name__` attribute.

```sh
Welcome to rshell. Use Control-D (or the exit command) to exit rshell.
/home/bradenm/Documents/projects/contrib/lib-python> cp ./blynk
./blynklib.egg-info/  ./blynklib.py         ./blynktimer.py       
/home/bradenm/Documents/projects/contrib/lib-python> cp ./blynktimer.py /pyboard/lib/bbtimer.py
/home/bradenm/Documents/projects/contrib/lib-python> repl
Entering REPL. Use Control-X to exit.
>
MicroPython v1.11-47-g1a51fc9dd on 2019-06-18; ESP32 module with ESP32
Type "help()" for more information.
>>> 
>>> import main
{'1_timer': 'Running', '0_timer': 'Running', '2_timer': 'Running'}
>>> main.start()
Hi There
Hi There 2
Hi There
Hi There
{'1_timer': 'Running', '0_timer': 'Running', '2_timer': 'Running'}
Hi There 2
Hi There
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "main.py", line 21, in start
  File "/lib/bbtimer.py", line 86, in run
  File "/lib/bbtimer.py", line 13, in <lambda>
KeyboardInterrupt: 
>>> 
```

[Here is a pastebin of main.py](https://pastebin.com/rKgUqBCD)